src/sentry/web/frontend/group_tag_export.py

```diff
@@ -2,107 +2,41 @@
-
-    return wrapped
-
-
 class GroupTagExportView(ProjectView, CsvMixin, EnvironmentMixin):
     required_scope = "event:read"
 
     def get_header(self, key):
-        if key == "user":
-            return self.get_user_header()
-        return self.get_generic_header()
+        return tuple(IssuesByTagProcessor.get_fields(key))
 
     def get_row(self, item, key):
-        if key == "user":
-            return self.get_user_row(item)
-        return self.get_generic_row(item)
-
-    def get_generic_header(self):
-        return ("value", "times_seen", "last_seen", "first_seen")
-
-    def get_generic_row(self, item):
-        return (
-            item.value,
-            item.times_seen,
-            item.last_seen.strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
-            item.first_seen.strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
-        )
-
-    def get_user_header(self):
-        return (
-            "value",
-            "id",
-            "email",
-            "username",
-            "ip_address",
-            "times_seen",
-            "last_seen",
-            "first_seen",
-        )
-
-    def get_user_row(self, item):
-        euser = item._eventuser
-        return (
-            item.value,
-            euser.ident if euser else "",
-            euser.email if euser else "",
-            euser.username if euser else "",
-            euser.ip_address if euser else "",
-            item.times_seen,
-            item.last_seen.strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
-            item.first_seen.strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
-        )
+        fields = IssuesByTagProcessor.get_fields(key)
+        item_dict = IssuesByTagProcessor.serialize_issue(item, key)
+        return (item_dict[field] for field in fields)
 
     def get(self, request, organization, project, group_id, key):
-        try:
-            # TODO(tkaemming): This should *actually* redirect, see similar
-            # comment in ``GroupEndpoint.convert_args``.
-            group, _ = get_group_with_redirect(
-                group_id, queryset=Group.objects.filter(project=project)
-            )
-        except Group.DoesNotExist:
-            raise Http404
-
-        if tagstore.is_reserved_key(key):
-            lookup_key = u"sentry:{0}".format(key)
-        else:
-            lookup_key = key
 
+        # If the environment doesn't exist then the tag can't possibly exist
         try:
             environment_id = self._get_environment_id_from_request(request, project.organization_id)
         except Environment.DoesNotExist:
-            # if the environment doesn't exist then the tag can't possibly exist
             raise Http404
 
-        # validate existence as it may be deleted
         try:
-            tagstore.get_tag_key(project.id, environment_id, lookup_key)
-        except tagstore.TagKeyNotFound:
+            processor = IssuesByTagProcessor(
+                project_id=project.id, group_id=group_id, key=key, environment_id=environment_id
+            )
```

Comment:
Again, the constructor cuts out _all_ of the processing code that was below, skipping right to the CSV response